// @progress/kendo-buttons-vue-wrapper/dist/es/KendoToolBar/KendoToolBarMixin.js
'use strict';

Object.defineProperty(exports, "__esModule", {
    value: true
});

var _kendoBaseComponentsVueWrapper = require('@progress/kendo-base-components-vue-wrapper');

var _ToolBarProps = require('./ToolBarProps');

var _ToolBarProps2 = _interopRequireDefault(_ToolBarProps);

var _KendoToolBarItem = require('./KendoToolBarItem');

var _KendoToolBarItem2 = _interopRequireDefault(_KendoToolBarItem);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

exports.default = {
    props: _ToolBarProps2.default,
    emits: {
        'kendowidgetready': null
    },
    mixins: [_kendoBaseComponentsVueWrapper.KendoBaseComponent],
    mounted: function mounted() {
        this._resolveChildren();

        kendo.jQuery(this.$el).kendoToolBar(this.widgetOptions);

        if (this.v3) {
            this.ready();
            this.$emit('kendowidgetready', this.kendoWidget());
        } else {
            this.$emit('kendowidgetready', this.kendoWidget());
        }
    },

    methods: {
        kendoWidget: function kendoWidget() {
            return kendo.jQuery(this.$el).getKendoToolBar();
        },
        _resolveChildren: function _resolveChildren() {
            this.resolveChildren('items', _KendoToolBarItem2.default.name);
        }
    }
};
