// @progress/kendo-dialog-vue-wrapper/dist/npm/KendoDialog/KendoDialogMixin.js
'use strict';

Object.defineProperty(exports, "__esModule", {
    value: true
});

var _kendoBaseComponentsVueWrapper = require('@progress/kendo-base-components-vue-wrapper');

var _DialogProps = require('./DialogProps');

var _DialogProps2 = _interopRequireDefault(_DialogProps);

var _KendoDialogAction = require('./KendoDialogAction');

var _KendoDialogAction2 = _interopRequireDefault(_KendoDialogAction);

var _Animation = require('./Animation');

var _Animation2 = _interopRequireDefault(_Animation);

var _Messages = require('./Messages');

var _Messages2 = _interopRequireDefault(_Messages);

var _Modal = require('./Modal');

var _Modal2 = _interopRequireDefault(_Modal);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

exports.default = {
    props: _DialogProps2.default,
    emits: {
        'kendowidgetready': null
    },
    mixins: [_kendoBaseComponentsVueWrapper.KendoBaseComponent, _Animation2.default, _Messages2.default, _Modal2.default],
    mounted() {
        this._resolveChildren();

        kendo.jQuery(this.$el).kendoDialog(this.widgetOptions);

        if (this.v3) {
            this.ready();
            this.$emit('kendowidgetready', this.kendoWidget());
        } else {
            this.$emit('kendowidgetready', this.kendoWidget());
        }
    },
    methods: {
        kendoWidget() {
            return kendo.jQuery(this.$el).getKendoDialog();
        },
        _resolveChildren() {
            this.resolveChildren('actions', _KendoDialogAction2.default.name);
        },
        toggleClasses() {}
    }
};
