// @progress/kendo-dialog-vue-wrapper/dist/npm/index.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.DialogInstaller = exports.KendoDialogInstaller = exports.DialogAction = exports.KendoDialogAction = exports.Dialog = exports.KendoDialog = undefined;

var _KendoDialog = require('./KendoDialog');

var _KendoDialog2 = _interopRequireDefault(_KendoDialog);

var _KendoDialogAction = require('./KendoDialog/KendoDialogAction');

var _KendoDialogAction2 = _interopRequireDefault(_KendoDialogAction);

var _kendoDialogInstaller = require('./kendo-dialog-installer');

var _kendoDialogInstaller2 = _interopRequireDefault(_kendoDialogInstaller);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

exports.KendoDialog = _KendoDialog2.default;
exports.Dialog = _KendoDialog2.default;
exports.KendoDialogAction = _KendoDialogAction2.default;
exports.DialogAction = _KendoDialogAction2.default;
exports.KendoDialogInstaller = _kendoDialogInstaller2.default;
exports.DialogInstaller = _kendoDialogInstaller2.default;
