// @progress/kendo-vue-buttons/dist/npm/toolbar/tools/ToolbarSpacer.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.ToolbarSpacerVue2 = exports.ToolbarSpacer = void 0; // @ts-ignore

var Vue = require("vue");

var allVue = Vue;
var gh = allVue.h;
var isV3 = allVue.version && allVue.version[0] === '3';
/**
 * @hidden
 */

var ToolbarSpacerVue2 = {
  name: 'KendoToolbarItem',
  // @ts-ignore
  setup: !isV3 ? undefined : function () {
    var v3 = !!isV3;
    return {
      v3: v3
    };
  },
  render: function render(createElement) {
    var h = gh || createElement;
    return h("span", {
      "class": "k-spacer"
    });
  }
};
exports.ToolbarSpacerVue2 = ToolbarSpacerVue2;
/**
 * Represents the [Kendo UI for Vue ToolbarSpacer component]({% slug overview_toolbar %}).  A spacer element for the sub-elements of the Toolbar.
 *
 *  ```jsx
 *  <template>
 *    <Toolbar>
 *    <ToolbarItem>
 *      <ButtonGroup>
 *        <Button :icon="'bold'" :title="'Bold'" :togglable="true" />
 *        <Button :icon="'italic'" :title="'Italic'" :togglable="true" />
 *        <Button :icon="'underline'" :title="'Underline'" :togglable="true" />
 *      </ButtonGroup>
 *    </ToolbarItem>
 *    <ToolbarSpacer />
 *    <ToolbarItem>
 *      <SplitButton :text="'Insert'" :items="splitItems"> </SplitButton>
 *    </ToolbarItem>
 *    <ToolbarSeparator />
 *    <ToolbarItem>
 *      <Button :icon="'cut'" :title="'Cut'"> Cut </Button>
 *    </ToolbarItem>
 *  </Toolbar>
 *  </template>
 *  ```
 */

var ToolbarSpacer = ToolbarSpacerVue2;
exports.ToolbarSpacer = ToolbarSpacer;
