// @progress/kendo-vue-dialogs/dist/es/DialogActionsBar.js
// @ts-ignore
import * as Vue from 'vue';
var allVue = Vue;
var gh = allVue.h;
var isV3 = allVue.version && allVue.version[0] === '3';
import { getDefaultSlots } from '@progress/kendo-vue-common';
/**
 * @hidden
 */

var DialogActionsBarVue2 = {
  name: 'DialogActionsBar',
  props: {
    layout: {
      type: String,
      default: 'stretched'
    },
    buttonGroupClass: {
      type: Boolean,
      default: true
    }
  },
  // @ts-ignore
  setup: !isV3 ? undefined : function () {
    var v3 = !!isV3;
    return {
      v3: v3
    };
  },
  computed: {
    wrapperClasses: function wrapperClasses() {
      var layout = this.$props.layout;
      return {
        'k-dialog-buttongroup': this.buttonGroupClass,
        'k-actions': true,
        'k-hstack': true,
        'k-justify-content-start': layout === 'start',
        'k-justify-content-center': layout === 'center',
        'k-justify-content-end': layout === 'end',
        'k-justify-content-stretch': layout === 'stretched'
      };
    }
  },
  // @ts-ignore
  render: function render(createElement) {
    var h = gh || createElement;
    var defaultSlot = getDefaultSlots(this);
    return h("div", {
      "class": this.wrapperClasses
    }, [defaultSlot]);
  }
};
/**
 * @hidden
 */

var DialogActionsBar = DialogActionsBarVue2;
export { DialogActionsBar, DialogActionsBarVue2 };
